porcupine-core/src/Data/Locations/VirtualFile.hs

```diff
@@ -114,13 +114,19 @@ instance HasDefaultMappingRule (VirtualFile a b) where
 
 -- For now, given the requirement of PTask, VirtualFile has to be a Monoid
 -- because a Resource Tree also has to.
+-- TODO: explain the meaning of (<>). From the implementation,
+-- computing the union of serials and ignoring the location seems pretty
+-- arbitrary.
 instance Semigroup (VirtualFile a b) where
   VirtualFile p u m i d b s <> VirtualFile _ _ _ _ _ _ s' =
     VirtualFile p u m i d b (s<>s')
+-- TODO: It doesn't look like this instance satisfies the identity
+-- laws of Monoid. Comment here what's the plan to address this?
 instance Monoid (VirtualFile a b) where
   mempty = VirtualFile [] SingleLayerRead True def Nothing Nothing mempty
 
 instance Profunctor VirtualFile where
+  -- TODO: explain why the LayeredReadScheme of the input is ignored?
```

Comment:
Yep, that's because if the input is changed, we cannot get a proof that it is still a Monoid, so we cannot keep its LayeredReadScheme.